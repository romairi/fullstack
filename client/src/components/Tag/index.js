import React from 'react';
import _ from 'lodash';
import './index.scss';

function Tag({id, children, onDeleteClicked=_.noop}) {
    return (
        <div className="tag_item">
            {children}
            <span className="tag_item_delete_button" onClick={() => onDeleteClicked(id)}>X</span>
        </div>
    );
}

export default Tag;
