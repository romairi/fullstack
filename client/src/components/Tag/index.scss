.tag_items {
  display: inline-flex;
  margin-right: 5px;
  margin-top: 5px;

  .chip_tag_item {
    display: flex;
    justify-content: start;
    background: #e9e8e8;
    flex-wrap: wrap;
    margin-bottom: 10px;
    margin-right: 5px;

    &:last-child {
      margin-right: 0;
    }
  }
}
