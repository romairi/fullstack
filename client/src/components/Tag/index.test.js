import React from 'react';
import renderer from 'react-test-renderer';
import Tag from './index';

describe('Tag', () => {
    describe('render', () => {
        it('should render the component - without children', () => {
            const onDeleteClicked = jest.fn();
            const component = renderer.create(
                <Tag id='SOME_ID' onDeleteClicked={onDeleteClicked} />,
            );
            expect(component.toJSON()).toMatchSnapshot();
        });
        it('should render the component - with children', () => {
            const onDeleteClicked = jest.fn();
            const component = renderer.create(
                <Tag id='SOME_ID' onDeleteClicked={onDeleteClicked} ><div>CHILDREN</div></Tag>,
            );
            expect(component.toJSON()).toMatchSnapshot();
        });
    });
    describe('onClick', () => {
        const id = 'SOME_ID';
        const onDeleteClicked = jest.fn();
        const component = renderer.create(
            <Tag id={id} onDeleteClicked={onDeleteClicked} />,
        );
        const spanElement = component.root.findByType('span');
        spanElement.props.onClick();
        expect(onDeleteClicked).toHaveBeenCalledWith(id);
    });
});
