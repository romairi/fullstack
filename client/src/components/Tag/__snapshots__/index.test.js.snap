// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag render should render the component - without children 1`] = `
<div
  className="tag_item"
>
  <span
    className="tag_item_delete_button"
    onClick={[Function]}
  >
    X
  </span>
</div>
`;

exports[`Tag render should render the component - without children 2`] = `
<div
  className="tag_item"
>
  <div>
    CHILDREN
  </div>
  <span
    className="tag_item_delete_button"
    onClick={[Function]}
  >
    X
  </span>
</div>
`;
